primo/opt_model/result_parser.py

```diff
@@ -0,0 +1,776 @@
+        """
+        col_name = self._col_names.dist_to_road
+        self._check_column_exists(col_name)
+        return calculate_average(self.well_data.data, self._col_names.dist_to_road)
+
+    @property
+    def num_unique_owners(self):
+        """
+        Returns the number of well owners for a project
+        """
+        col_name = self._col_names.operator_name
+        self._check_column_exists(col_name)
+        return len(set(self.well_data[col_name].values))
+
+    @property
+    def impact_score(self):
+        """
+        Returns the average priority score for the project
+        """
+        if not hasattr(self._col_names, "priority_score"):
+            raise AttributeError(
+                "The priority score has not been computed for the Well Data"
+            )
+        return self.well_data[self._col_names.priority_score].mean()
+
+    @property
+    def accessibility_score(self):
+        """
+        Returns the accessibility score and the total weight of the accessibility score for a project
+        """
+        names_attributes = [i for i in dir(self) if "eff_score" in i]
+        names_attributes_accessibility = [
+            name
+            for name in names_attributes
+            if any([name2 in name for name2 in self.accessibility_attr])
+        ]
+        if len(names_attributes_accessibility) == 0:
+            return None
+        total_weight = sum(
+            int(name.split("_0_")[-1]) for name in names_attributes_accessibility
+        )
+        return total_weight, sum(
+            getattr(self, attribute) for attribute in names_attributes_accessibility
+        )
+
+    def _add_distance_to_centroid_col(self):
+        """
+        Adds the distance to centroid column to the well data
+        """
+        # I am assuming here that apply returns the data in the corresponding order
+        self.well_data.add_new_column_ordered(
+            ["dist_centroid", "Distance to Centroid [miles]"],
+            self.well_data.data.apply(
+                lambda row: get_distance(
+                    (row[self._col_names.latitude], row[self._col_names.longitude]),
+                    self.centroid,
+                ),
+                axis=1,
+            ).values,
+        )
+
+    def get_max_val_col(self, col_name) -> float:
+        """
+        Returns the maximum value of a column
+
+        Parameters
+        ----------
+        col_name : str
+            column name
+
+        Returns
+        -------
+        float
+            the maximum value of the column
+
+        """
+        return self.well_data[col_name].max()
+
+    def update_efficiency_score(self, value: Union[int, float]):
+        """
+        Updates the efficiency score of a project
+
+        Parameters
+        ----------
+        value : Union[int, float]
+            The value to update the efficiency score with
+        """
+        if len(self.well_data) == 1:

```

Comment:
This was a topic brought up in the last code dev/review meeting. The way that the efficiency calculations are currently computed, a project with only 1 well project will have an inaccurate efficiency score computed because the range of e.g., age for a project with a single well is 0, which is normalized by the max(age of all wells) - min(age of all wells), so it lies outside of the range of normalization and will therefore give a value outside of the intended range (if the range is 0-10, then it will either give a value >10 or <0). This is also the case for projects with wells that have homogenous data for a given characteristic. We discussed overcoming this by using the infinity norm. The changes are a little too substantial to be included in this PR...but a future PR that addresses the efficiency calculation should take care of this.

Reply:
yes, makes sense. Future PR it is